src/main/java/com/google/sps/servlets/Input.java

```diff
@@ -0,0 +1,69 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/input")
+public class Input extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+   
+    Query<Entity> query = Query.newEntityQueryBuilder()
+    .setKind("Listing")
+    .build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    List<Task> lists = new ArrayList<>();
+    while (results.hasNext()) {
+      Entity entity = results.next();
+
+      String name = entity.getString("name");
+      long location = entity.getLong("location");
+      String item = entity.getString("item");
+      long number = entity.getLong("number");
+
+      Task list = new Task(name, location, item, number);
+      lists.add(list);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(lists));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String text = request.getParameter("listing");
+    
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Listing");
+    FullEntity listingEntity = Entity.newBuilder(keyFactory.newKey())
+      .set("name", name)
+      .set("location", location)
+      .set("item", item)
+      .set("number", number)
+      .build();
+    datastore.put(listingEntity);
+
+    response.sendRedirect("/index.html");
+  }
+}
```

Comment:
Since your updated code is located in src/main/java/com/google/sps/servlets/DisplayServlet.java, you can delete this file